core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -125,13 +122,7 @@ static class ObjectStoreLocationProvider implements LocationProvider {
     private static String dataLocation(Map<String, String> properties, String tableLocation) {
       String dataLocation = properties.get(TableProperties.WRITE_DATA_LOCATION);
       if (dataLocation == null) {
-        dataLocation = properties.get(TableProperties.OBJECT_STORE_PATH);
-        if (dataLocation == null) {
-          dataLocation = properties.get(TableProperties.WRITE_FOLDER_STORAGE_LOCATION);
-          if (dataLocation == null) {
-            dataLocation = String.format("%s/data", tableLocation);
-          }
-        }
```

Comment:
I'm not sure about this. I understand we want to deprecate these properties, but wouldn't this break compatibility in terms of the data location to write to for existing tables which have these older properties set? 

This may be intentional desire with an upgraded 1.0 Iceberg library and an older table, but it basically forces users to have to update their table properties on their tables to use 1.0 Iceberg library and have their existing behavior which does not seem entirely right to me.

Reply:
@amogh-jahagirdar thanks for the feedback!
I had the same concern of dropping these properties that it does have an effect on existing tables. I created this PR so that we can discuss. I wasn't sure that a major release bump allows us to make such a big break in the API and on existing tables.

I'm wondering if even a major version bump is not enough to drop these properties then is there a point of annotating these properties as deprecated. In case there isn't a scenario where we can drop them I think we can remove that annotation. Should I go forward and do so?
Some of them are even said to be removed in 0.14 and are still around that I find misleading.
Another idea would be that we leave them as they are now but setting it should be prevented in 1.0.0 onwards. E.g. it won't be allowed to create a table using those properties, and we could also reject setting these for existing tables.

I'd appreciate to hear your opinion.

Tagging @flyrain as the author of the patch that introduced those annotations:
https://github.com/apache/iceberg/commit/12e30ebaeb6dc0ff7e6bb01c0d9f8ae8a5f06ddf#diff-5620e28d3a7d13259b12e251b9eca2f59c66fd3ab96d6b62c36062f1f5ac0ca1

Reply:
I don't want to make it more complex, but does it make sense to have a tool to migrate tables from old versions to 1.0? We can replace the deprecated properties with the tool.